testdata/BUILD

```diff
@@ -796,18 +796,6 @@ java_image(
     runtime_deps = [":java_bin_as_lib"],
 )
 
-java_binary(
-    name = "java_runfiles_binary",
-    srcs = ["Runfiles.java"],
-    data = [
-        "foo",
-    ],
-    main_class = "examples.images.Runfiles",
-    deps = [
-        "@bazel_tools//tools/runfiles:java-runfiles",
-    ],
-)
-
 java_image(
     name = "java_runfiles_image",
```

Comment:
could you please add a test for java_runfiles_image in container/image_test.py

Reply:
In the process of working on this.

I'm running into some problems with layer 1 -- it looks like it passes java_files in as lambda, and that is also picking up the data files. I'm going to fix it by having a version of java_files that doesn't include the data dependencies, but it raised a question:

*should* the java_runfiles_as_lib case have the runfiles in the top layer, or the dependency layer?

Reply:
good question. I personally do not have enough strong arguments in favor of either. From a conceptual point of view it might seem to make sense to put in the dependency layer (iiuc). Do you have any arguments to put it in top or dep layer or can imagine any trade-offs this would bring?

Reply:
I agree that it makes the most sense to put it in the dependency layer. However, trying to make that work I ran into problems, since that makes use of functionality from lang/image.bzl, which wasn't easily extendable to support excluding the JDK files.

My inclination is to leave it as is -- data files of dependencies in the top layer -- to get things working for the time being. The disadvantage is that a change in dependencies may require rebuilding both the dependency layer and the top layer (I think), but I'm not sure of the cost of that. It also means that every time the top layer changes, the data dependencies need to be re-tar'd, etc. Again, likely not a huge problem, and something that seems better to fix in the future by adding appropriate exclusion mechanisms to the rules that build the dep layer.

Reply:
sgtm, could you leave a comment in the code about this decision so that we dont loose track it.

Reply:
Done